lib/Browser.js

```diff
@@ -33,6 +33,9 @@ class Browser extends EventEmitter {
     this._closeCallback = closeCallback || new Function();
     /** @type {Map<string, Target>} */
     this._targets = new Map();
+    this._connection.on('ws.closed', closed => {
+      this.disconnect();
```

Comment:
no need to run `this.disconnect()` here - you only need to emit the event.

Reply:
done